common/src/main/java/com/graphicsfuzz/common/util/PipelineInfo.java

```diff
@@ -74,17 +74,24 @@ public PipelineInfo(String string) {
   public void addUniform(String name, BasicType basicType,
       Optional<Integer> arrayCount, List<? extends Number> values) {
     assert isLegalUniformName(name);
-    JsonObject info = new JsonObject();
-    info.addProperty("func", PipelineInfo.get(basicType, arrayCount.isPresent()));
-    JsonArray jsonValues = new JsonArray();
-    for (Number n : values) {
-      jsonValues.add(n);
-    }
-    info.add("args", jsonValues);
-    if (arrayCount.isPresent()) {
-      info.addProperty("count", arrayCount.get());
+    if (!dictionary.has(name)) {
```

Comment:
I'm keen on the style of eliminating all erroneous cases first, to avoid nesting (avoiding the "pyramid of doom").  So I would prefer you to do:

if (dictionary.has(name)) {
   // throw the exception
}
// No else needed; do the non-exceptional case

Reply:
I guess I'm wrong here, though, since in the "else" you don't unconditionally throw an exception.

In that case I suggest you use "else if", to avoid "else { if { "

Reply:
I swapped them around anyway, and added clarifying comments